http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/GzipMaxInputIT.java

```diff
@@ -0,0 +1,126 @@
+package io.quarkus.ts.http.advanced;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+import jakarta.ws.rs.core.MediaType;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.Response;
+
+@QuarkusScenario
+public class GzipMaxInputIT {
```

Comment:
I'd rather see something like https://gist.github.com/rsvoboda/01538aa344524805e5a1eb3ad4ef28c0
(got excited during the meetings afternoon)

Test methods have self descriptive names, there is a JavaDoc description for the file, explanatory messages for the persona facing the test failure, extracted boilerplate logic so the tests are easily readable, more robust gziping method with try-with-resources to avoid heap memory occupation.

PASS: `mvn clean verify -Dit.test=GzipMaxInputIT -Dquarkus.platform.version=3.8.3 -Dreruns=0`
FAIL: `mvn clean verify -Dit.test=GzipMaxInputIT -Dreruns=0`